### DISCUSSION TITLE
Why is this grammar not working

### DISCUSSION THREAD

**Discussion by mnaderhirn** [eclipse-langium/langium#1593] [General]
[Playground](https://langium.org/playground?grammar=OYJwhgthYgBASgUwGaJIgdgY0QcXFDAFBGYAuIAngimpjgFxGwuwAUA5AN5cewDKABQCCAYQCi-APw1UIDPUQCRE6bA4BfDRwCUsAD7sOAHWPpkfbnyFjJMpHMXLbazboDcJVrLQLsiJm9vADEAS2AAV3QDWAAVMAAjABslQ2EEhPQAN1CwMlCAewwHGNxMNDAkz2ZWMMj0QKCWcwB9MkoAB0QAXg5kcKjESwZrFTtYc26AOWEAWXFq73jkgJqm1vau3rJElOHRlxlJmfnF1nTMxBy8wuKURvWUNs6ejkTLrP3nVSOUabmFl5WGUFOAkg8ghsXr1gOUwRwZBwRjIbD8Jn8ToCiGQ0BBQhhKt9JAwAPTGADOAGoSZ4cSA8QSkrBMQxYCSANotMAAWgAXsJuQAtAC67OMAHcWsKAFQ0oA&content=DoJwpgZg3gFghgGyQewL4Cgg)
I trying to implement the simple grammar in the playground and it is giving me errors, i dont understand. Could anyone have a look at it please?

**Comment by montymxb** [comment]
Hey @mnaderhirn . There's a couple things going on here that are issues. You don't need to explicitly handle whitespace (there's hidden terminals for that case, ex `hidden terminal WS:/\s+/;`), and you also need a way to declare more than one statement to have something to reference, like so:
```antlr
entry Reference:
    (
        references+=Refernence | // also allow declaring things you can reference, and collect them
        ('{{' Refernence '}}') |
        ('\\ref' '{' Reference '}')
    )*; // zero or more of these
```
For reference you can check out [this modified version of your grammar.](https://langium.org/playground?grammar=OYJwhgthYgBASgUwGaJIgdgY0QcXFDAFBGYAuIAngimpjgFxGwuwAUzrX6q62iAZwDUAXiS8M9RLAA%2BnLizYByAN4qlNCVNhKAvrqUBKWfIXKAOuZ4bVG8XX46Dh04YBUAbhJd7ISfyYFVgAxAEtgAFd0WVgAFTAAIwAbaRlYAEEEhPQAN1CwMlCAewxxGNxMNDAkr1MwyPRAoNgeAH0ySgAHRBElZHCoxBsGJR4RADl0gFkAUVqueOTEJqC2ju7eskSU4Y0xydn51kzsxDyC4tKUFYU1rp6lRNOc3ZaUCem571YKyXAkm7cFDte69YCVf5KAD8OhGMP2n1qAAtQgATVGYWBkNAQUIYaqwADqAGUGAB6czCMlebEgXH4pKwA4zBiwMkAbVaYAAtAAvdLcgBaAF12eYAO6tYVualAA&content=GYSw5grgTgpgXAAgPIGkBQaA6tgG9UC%2BQA) Hope that helps!
